test/MinimalDelegation.execute.t.sol

```diff
@@ -628,4 +628,103 @@ contract MinimalDelegationExecuteTest is TokenHandler, HookHandler, ExecuteFixtu
+            signedBatchedCall2, // the SignedBatchedCall struct
+            signature2
+        );
+
+        // Execute the multicall
+        vm.prank(address(signerAccount));
+        signerAccount.multicall(multicallData);
+
+        // Verify the results
+        assertEq(tokenA.balanceOf(address(receiver)), 1e18);
+        assertEq(tokenB.balanceOf(address(receiver)), 1e18);
+    }
+
+    /// forge-config: default.isolate = true
+    /// forge-config: ci.isolate = true
+    function test_multicall_gas() public {
+        // Create two separate batches of calls
+        Call[] memory calls1 = CallUtils.initArray();
+        Call[] memory calls2 = CallUtils.initArray();
+
+        calls1 = calls1.push(buildTransferCall(address(tokenA), address(receiver), 1e18)); // Transfer 1 tokenA
+        calls2 = calls2.push(buildTransferCall(address(tokenB), address(receiver), 1e18)); // Transfer 1 tokenB
+
+        TestKey memory p256Key1 = TestKeyManager.initDefault(KeyType.P256);
+        TestKey memory p256Key2 = TestKeyManager.initDefault(KeyType.P256);
+
+        vm.startPrank(address(signer));
+        signerAccount.register(p256Key1.toKey());
+        signerAccount.register(p256Key2.toKey());
+        vm.stopPrank();
+
+        // Create signed calls for each batch
+        BatchedCall memory batchedCall1 = CallUtils.initBatchedCall().withCalls(calls1).withShouldRevert(true);
+        SignedBatchedCall memory signedBatchedCall1 = CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall1)
+            .withNonce(DEFAULT_NONCE).withKeyHash(p256Key1.toKeyHash());
+        bytes memory signature1 = p256Key1.sign(signerAccount.hashTypedData(signedBatchedCall1.hash()));
+
+        BatchedCall memory batchedCall2 = CallUtils.initBatchedCall().withCalls(calls2).withShouldRevert(true);
+        SignedBatchedCall memory signedBatchedCall2 = CallUtils.initSignedBatchedCall().withBatchedCall(batchedCall2)
+            .withNonce(DEFAULT_NONCE + 1).withKeyHash(p256Key2.toKeyHash());
+        bytes memory signature2 = p256Key2.sign(signerAccount.hashTypedData(signedBatchedCall2.hash()));
+
+        // Create multicall data
+        bytes[] memory multicallData = new bytes[](2);
+        multicallData[0] = abi.encodeWithSelector(
+            bytes4(keccak256("execute((((address,uint256,bytes)[],bool),uint256,bytes32),bytes)")),
+            signedBatchedCall1, // the SignedBatchedCall struct
+            signature1
+        );
+
+        multicallData[1] = abi.encodeWithSelector(
+            bytes4(keccak256("execute((((address,uint256,bytes)[],bool),uint256,bytes32),bytes)")),
+            signedBatchedCall2, // the SignedBatchedCall struct
+            signature2
+        );
+
+        // Execute the multicall
+        vm.prank(address(signerAccount));
+        signerAccount.multicall(multicallData);
+        vm.snapshotGasLastCall("multicall");
+    }
```

Comment:
I'd want to see a test showing that the multicall passes on the original call's context:
- multicall with `execute(Signed)` and `execute()` by p256 key, first one succeeds but second one fails because the key is not root user
- multicall with root key can call multiple functions on address(this)